server/plugins/client/bundle-handler-factory.js

```diff
@@ -0,0 +1,54 @@
+module.exports.createHandler = createHandler
+
+var fs = require('fs')
+
+var log = require('npmlog')
+var bundleClient = require('./bundle')
+
+function createHandler (hoodieClientPath, bundleTargetPath, bundleConfig) {
+  var bundlePromise
+
+  return function (request, reply) {
+    if (!bundlePromise) {
+      bundlePromise = new Promise(function (resolve, reject) {
+        log.silly('client', 'bundling')
+        bundleClient(hoodieClientPath, bundleTargetPath, bundleConfig, function (error, bundleBuffer, hasUpdate) {
+          if (error) {
+            return reject(error)
+          }
+
+          writeClientBundle(hasUpdate, bundleConfig.inMemory, bundleTargetPath, bundleBuffer)
+
+          resolve(bundleBuffer)
+        })
+      })
+    }
+
+    bundlePromise.then(function (buffer) {
+      reply(buffer).bytes(buffer.length).type('application/javascript')
+    }).catch(reply)
+  }
+}
+
+function writeClientBundle (hasUpdate, inMemory, bundleTargetPath, bundleBuffer) {
+  if (!hasUpdate) {
+    log.info('client', 'bundle is up to date')
+    return
+  }
+
+  if (inMemory) {
+    log.silly('client', 'running in memory, not writing bundle to ' + bundleTargetPath)
+    return
+  }
+
+  log.info('client', 'bundle is out of date')
+  log.silly('client', 'writing bundle to ' + bundleTargetPath)
+  fs.writeFile(bundleTargetPath, bundleBuffer, function (error) {
+    if (error) {
+      log.warn('client', 'could not write to ' + bundleTargetPath + '. Bundle cannot be cached and will be re-generated on server restart')
+      return
+    }
+
+    log.info('client', 'bundle written to ' + bundleTargetPath)
+  })
+}
```

Comment:
Haha, I was actually thinking about doing this but then thought it is not that complex to test right now.
But of course, it would make things even easier! Shall I do it within this PR or open another issue after we are done here?